poglink/config.py

```diff
@@ -41,8 +42,12 @@ def setup_config(args, default_config=DEFAULT_CONFIG):
     logger.debug(f"Setting up configuration. Checking for config file: {config_path}")
 
     if os.path.exists(config_path):
-        with open(os.path.expanduser(config_path)) as f:
-            config_from_file = yaml.safe_load(f)
+        try:
+            with open(os.path.expanduser(config_path)) as f:
+                config_from_file = yaml.safe_load(f)
+        except Exception as e:
+            logger.error(f"Problem reading configuration file at {config_path}: {e}")
+            raise ConfigReadError(e) from e
```

Comment:
I've added this in to address your previous concern, but I just wanted to point out that we don't actually have any unit tests set up to test that this error is raised (I did add the case where it's being loaded via `BotConfig.from_file()` though). 

Kinda tricky to implement too becaus of the way I've got [this](https://github.com/FM-17/poglink/blob/5236a6a9174f44d7beeb2a24bb77a041454cf150/tests/unit/test_config.py#L39-L46) fixture set up for the existing file-loading test assertion. 

We technically now have two methods of reading config from file (as part of the `poglink.config.setup_config()` function and as part of the `poglink.bot.BotConfig.from_file()` classmethod. We're currently only using the former in the application code.

When we address #41, I'd like to consolidate this so all config is handled within the `BotConfig` class for uniformity, and we can just make sure the test coverage is good when we get to that.